### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by andrewmil**
Added authentication with flask-security and user administration with flask-admin

### CODE REVIEW

File: console/authentication.py [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+    session.permanent = True
+    app.permanent_session_lifetime = timedelta(minutes=settings.CONSOLE_LOGIN_TIMEOUT)
+
+
+class UserAdmin(sqla.ModelView):
+    column_exclude_list = ('password',)
+    form_excluded_columns = ('password',)
+    column_auto_select_related = True
+
+    def is_accessible(self):
```
Comment (necrophonic) [inline console/authentication.py]: Doesn't use `self` so could be function rather than a method

File: console/authentication.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+        return flask_security.core.current_user.has_role('Admin')
+
+    def scaffold_form(self):
+        form_class = super(UserAdmin, self).scaffold_form()
+        form_class.password2 = PasswordField('New Password')
+
+        return form_class
+
+    def on_model_change(self, form, model, is_created):
```
Comment (necrophonic) [inline console/authentication.py]: Doesn't use `self` so could be function rather than a method

File: console/authentication.py (hunk 1 above)
Comment (shahi645) [inline console/authentication.py]: can be a static method

File: console/authentication.py (hunk 2 above)
Comment (shahi645) [inline console/authentication.py]: can also be a static method

**Review by shahi645 — CHANGES_REQUESTED** [review]
Changes listed in authentication.py

**Review by shahi645 — COMMENTED** [review]
changes look good!

**Review by shahi645 — APPROVED** [review]

**Pull request merged**